src/components/PersonLink/PersonLink.tsx

```diff
@@ -0,0 +1,78 @@
+import classNames from 'classnames';
+import { Person } from '../../types/Person';
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import { useCallback, useMemo } from 'react';
+
+type Props = {
+  person: Person;
+  sortedPeople: Person[];
+};
+
+export const PersonLink: React.FC<Props> = ({ person, sortedPeople }) => {
+  const [searchParams] = useSearchParams();
+  const { personSlug } = useParams();
+
+  const selectedPerson = useMemo(() => {
+    return sortedPeople.find(p => p.slug === personSlug);
+  }, [personSlug, sortedPeople]);
+
+  const getPersonByName = useCallback(
+    (name: string | null) => {
+      return sortedPeople.find(p => p.name === name);
+    },
+    [sortedPeople]
+  );
+
+  const mother = getPersonByName(person.motherName);
+  const father = getPersonByName(person.fatherName);
+
+  return (
+    <tr
+      key={person.slug}
+      data-cy="person"
+      className={classNames({
+        'has-background-warning': selectedPerson?.slug === person.slug,
+      })}
+    >
+      <td>
+        <Link
+          className={classNames({ 'has-text-danger': person.sex === 'f' })}
+          to={{
+            pathname: `${person.slug}`,
```

Comment:
The `pathname` for the `Link` component is set to `${person.slug}`. Ensure that this path is correctly handled in your routing configuration to display the person's details.